### PR TITLE
Add training script

### PR DESCRIPTION
**Pull request by anasneji2002**
# This PR includes:
 - Adding a script to train the models to be used in a cron job later
 - Add some utilities functions / modules to be used later / in the script
 - updated the .env.example and the train pipeline accordingly

This script and the solution in general was designed in a way that it would solely depend on the environment (only the script does, the utility methods don't). i.e if I for some reason want to no longer use LSTM and use another model instead (as long as the model is implemented and uses the same data as the LSTM) we would only need to change the env var MODEL_NAME and delete the azure ml model and run the pipeline manually : WE WONT NEED TO DO ANY CODE CHANGES WHATSOEVER (the same applies to the window size)
 
# What does the script do exactly (for each stock):
 - We fetch the csv file if existing
 - We get the model / scaler (for now we just instantiate them and later on we will integrate azure ML)
 - Get the model's last trained date (a new property added to the IModel abstract class) , this date corresponds to the date of the latest row on which the model was trained
 - if the date exists we check for rows with later dates and only use them for training, else we use the whole dataset to train the model and fit the scaler
 - after training we save the model and scaler (not implemented yet)
 
# What's left:
 - Implement the Azure ML integration part cc @ramirachdi 
 - potentially add mechanisms to check for regressions in model accuracy and send alerts

### CODE REVIEW

**Comment by anasneji2002** [comment]
the failing test is related to web/backend module (unrelated to my changes), @ramirachdi can you check please ?

**Review by ahmedrezgui — APPROVED** [review]
LGTM

File: src/prediction_model/models/ARIMAModel.py [hunk 1]
```diff
@@ -3,16 +3,18 @@
...
+        super().__init__(stock_name)
         pass
     
-    def train(self, features, targets):
+    def train(self, features, targets, last_trained_date):
         # for ARIMA model there isn't a separation between features and targets so we pass all the data in the features param
         features = features.flatten()
         print(f"features shape: {features.shape}")
         model = ARIMA(features, order=(2,1,5)) # the order was found when running auto_arima method which generated the best orders
         model_fit = model.fit()
```
Comment (neirezcher) [inline src/prediction_model/models/ARIMAModel.py:15]: Why don't you assign a new value to the attribute directly, like self.model = model.fit()? Is there a specific reason for avoiding that?

**Review by neirezcher — APPROVED** [review]
ellerine sağlık! çok iyi bir iş çıkardın 😉

File: src/prediction_model/models/ARIMAModel.py (hunk 1 above)
Comment (anasneji2002) [reply src/prediction_model/models/ARIMAModel.py:15]: this is old code (not related to the PR, but i don't know : maybe for readibility but not sure tbh)
will make sure to fix it in another pr

**Comment by anasneji2002** [comment]
merging the PR (will check the test afterwards since the tests pass locally)
